File: C/1653. Minimum Deletions to Make String Balanced.c
int minimumDeletions(char* s) {
    int b_before = 0, a_after = 0, result = 0;

    int i = 0;
    while (true) {
        if (*(s + i) == 'a')
            a_after++;
        else if (*(s + i) == '\0')
            break;

        i++;
    }

    result = a_after - (*s == 'a' ? 1 : 0);

    i = 0;
    while (true) {
        if (*(s + i) == 'a')
            a_after--;
        else if (*(s + i) == 'b')
            b_before++;
        else
            break;

        if (a_after + b_before < result)
            result = a_after + b_before;

        i++;
    }

    return result;
}

File: Python/1653. Minimum Deletions to Make String Balanced.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        b_before, a_after = 0, s.count("a")
        result = a_after - (1 if s[0] == "a" else 0)

        for c in s:
            if c == "a": a_after -= 1
            else: b_before += 1

            result = min(result, a_after + b_before)

        return result
